vue/components/ui/molecules/search-persistent/save-filter-modal.vue

```diff
@@ -95,25 +121,148 @@ export const SaveFilterModal = {
+        builtTenancies() {
+            const builtTenancies = {};
+
+            this.tenancyItems.forEach(item => {
+                if (item.value === "user") builtTenancies[item.value] = this.userTenancy;
+                else {
+                    builtTenancies[item.value] = {
+                        choices: [],
+                        selectedValue: null,
+                        selectVisible: false
+                    };
+                }
+            });
+
+            return builtTenancies;
         }
     },
-    mounted: async function() {},
+    mounted: async function() {
+        this.tenancies = this.builtTenancies;
+
+        // Getting brands, channels and factories
+        this.tenancies.brand.choices = await this.getBrands();
+        this.tenancies.channel.choices = await this.getChannels();
+        this.tenancies.factory.choices = await this.getFactories();
+
+        this.removeInvalidChoices();
+    },
     methods: {
+        async getBrands() {
+            // TODO fix me, I'm hardcoded and have 7 options
+            return [
+                { value: "brand_a", label: "Brand A" },
+                { value: "brand_b", label: "Brand B" },
+                { value: "brand_c", label: "Brand C" },
+                { value: "brand_d", label: "Brand D" },
+                { value: "brand_e", label: "Brand E" },
+                { value: "brand_f", label: "Brand F" },
+                { value: "brand_g", label: "Brand G" }
+            ];
+        },
+        async getChannels() {
+            // TODO fix me, I'm hardcoded and have 1 option
+            return [{ value: "channel_a", label: "Channel A" }];
+        },
+        async getFactories() {
+            // TODO fix me, I'm hardcoded and have 5 option
+            return [
+                { value: "factory_a", label: "Factory A" },
+                { value: "factory_b", label: "Factory B" },
+                { value: "factory_c", label: "Factory C" },
+                { value: "factory_d", label: "Factory D" },
+                { value: "factory_e", label: "Factory E" }
+            ];
+        },
+        removeInvalidChoices() {
+            for (let i = 0; i < this.tenancyItemsData.length; i++) {
+                if (!this.isTenancyChoiceValid(this.tenancyItemsData[i].value)) {
+                    this.tenancyItemsData.splice(i, 1);
+                    i--;
+                }
+            }
+        },
+        isTenancyChoiceSelected(value) {
+            return Boolean(this.tenancyCheckboxValuesData[value]);
+        },
+        isTenancyChoiceValid(value) {
+            return value === "user" || this.hasTenancyItems(value);
+        },
+        isTenancySelectValid(value) {
+            return this.hasTenancyItems(value) && this.hasRequiredItemsLength(value);
+        },
+        hasTenancyItems(value) {
+            return Boolean(this.tenancies[value] && this.tenancies[value].choices.length);
+        },
+        hasRequiredItemsLength(value) {

```

Comment:
Suggested renaming to avoid having to read the implementation to know what the functions do:
* `hasChoice(tenancy)`
* `hasChoices(tenancy)`

The names might not exactly be these, but something that makes it clear that the functions are almost the same.